### DISCUSSION TITLE
Add Option to Choose Auto-Complete Model and Separate Models for Different Features

### DISCUSSION THREAD

**Discussion by SohamTilekar** [zed-industries/zed#23373] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

I would like to introduce the `default_model` and `inline_alternatives` options. There should be an option to choose the auto-complete model, allowing users to select the fastest possible model provider/model, such as groq, which can provide completions in 1-2 seconds instead of using a latency model.

Additionally, I propose an optional feature to separate models for different functionalities like chat, inline, and autocomplete. This feature should include a precedence mechanism to determine which model to use first. For example, if the first model fails due to rate limits, the system will attempt to use the next model, and so on, until the last one. If all models fail, an error message will be displayed to the user.

### Zed Version and System Specs

request feature

### If applicable, add mockups / screenshots to help present your vision of the feature

![image](https://github.com/user-attachments/assets/040277c5-8be0-416a-98db-9b0fd96fc9c2)

**Comment by SohamTilekar** [comment]
> I dont know eng well so I tell the AI to reformulate my ans, ...

**Comment by fuguesoft** [comment]
My issue sounds similar:
The auto completion wreaks havoc with imports for example and changing the supposed settings don't seem to help.

Video of issue attached (same video):
https://files.catbox.moe/dlj8rb.webm
[Grabación de pantalla desde 2024-12-22 20-15-18.webm](https://github.com/user-attachments/assets/00cf24e4-50b7-42f6-ba85-6d737c1b6af2)
